src/core/search/sort_indices.cc

```diff
@@ -48,21 +75,31 @@ std::vector<ResultScore> SimpleValueSortIndex<T>::Sort(std::vector<DocId>* ids,
 template <typename T>
 bool SimpleValueSortIndex<T>::Add(DocId id, const DocumentAccessor& doc, std::string_view field) {
   auto field_value = Get(doc, field);
-  if (!field_value) {
+  if (!field_value.HasValue()) {
     return false;
   }
 
-  DCHECK_LE(id, values_.size());  // Doc ids grow at most by one
+  if (field_value.IsNullValue()) {
+    null_values_.insert(id);
+    return true;
+  }
+
   if (id >= values_.size())
     values_.resize(id + 1);
 
-  values_[id] = field_value.value();
+  values_[id] = std::move(field_value).Value();
   return true;
 }
 
 template <typename T>
 void SimpleValueSortIndex<T>::Remove(DocId id, const DocumentAccessor& doc,
                                      std::string_view field) {
+  auto it = null_values_.find(id);
+  if (it != null_values_.end()) {
```

Comment:
nit:
if (auto it = null_values_.find(id); it != null_values_.end()) 

it reduces variable scope